RDS 23 - course iterations 1

OoXLR8oO opened the pull request:
Blocked out a map to use for testing. Just a quick PR to make sure changes won't negatively affect dev branch.

jtbui20 committed:
feat: Implemented assembly def, initial clean up file structure

- Also added test & editor assembly def, rough organize namespace

jtbui20 committed:
feat: Moved types to single file

- Attempt at keeping "how data translates" away from "what the data is".
- Some code refactoring

jtbui20 committed:
feat: Move stunt system out

jtbui20 committed:
feat: Fix FMOD again

OoXLR8oO committed:
Basic track level blocked out

OoXLR8oO committed:
Merge branch 'dev' into rds-23-course-iterations-1

jtbui20 committed:
Merge branch 'dev' into critical/rds-211-code-cleanup

jtbui20 committed:
feat: Adjust assembly config

jtbui20 committed:
feat: Move scenes out to tests folder

jtbui20 committed:
fix: Dependencies reassign

jtbui20 committed:
feat: Remove other vehicles for now, code cleanup & fix dependencies

jtbui20 committed:
fix: Move more classes around

jtbui20 committed:
fix: Remove unused prefabs, fix visual scripting & imports

Sparoney committed:
Fix InputProcessor issue

jtbui20 committed:
Merge branch 'critical/rds-211-code-cleanup' into feature/rds-209-vehicle-movement-state-machine_extended

jtbui20 committed:
feat: Implement blackboard & state machine structures with test

jtbui20 committed:
feat: Implement data pipeline structure with tests

jtbui20 committed:
feat: Implement Vehicle Stats Scriptable Object

jtbui20 committed:
feat: Implement ground movement state & associated features

jtbui20 committed:
fix: Wheels no longer should clip into the ground when falling

jtbui20 committed:
feat: Prepare parent Player Input check instead

jtbui20 committed:
feat: adjust implementation & add upright feature

jtbui20 committed:
feat: prepare powerup system class

jtbui20 committed:
Merge branch 'dev' into rds-23-course-iterations-1

OoXLR8oO committed:
Updated map blockout

- Differentiated between track and buildings
- Added arrows at halfpipes to signal intention
- Converted whole map into prefab

OoXLR8oO committed:
Merge branch 'rds-23-course-iterations-1' of https://github.com/team-riders/project-riders into rds-23-course-iterations-1

AppelJoosh committed:
remove AddPowerup()

jtbui20 committed:
feat: Implement Airborne Temporary state, fix common classes, add feats

jtbui20 committed:
feat: [broken] Added grinding state and implement related features

jtbui20 committed:
feat: Attempt fix exit feature

jtbui20 committed:
feat: Implement grinding to working state

jtbui20 committed:
Merge remote-tracking branch 'origin/dev' into feature/rds-209-vehicle-movement-state-machine_extended

jtbui20 committed:
fix: bug where overshooting still allowed grinding

jtbui20 committed:
fix: Fix rotation to work alongside curve

jtbui20 committed:
fix: Signify optional reference

jtbui20 committed:
format: Import structures

jtbui20 commented:
1. please move the map to a prefab. 
2. please also make the map legible, i.e. use colour to distinguish what's a building, what surface I should stay on, where should I go, etc.
3. please fill in that gap for the jump for now, we don't have a respawn systen ready nor planned
4. arrow symbol on the half pipe to demonstrate intention. the first time I went I thought I had to go backwards
5. seal off some areas where the player may yeet themselves off
6. jack up the speed a ton

jtbui20 commented:
Recorded work as of 26th April
![image](https://github.com/user-attachments/assets/3a256a97-94fd-4b4e-a7cb-0875d92ecd6c)

OoXLR8oO committed:
Added walls to stop player from yeeting themselves off the map

OoXLR8oO committed:
Fixed location transform for track

Before: The 'Track' child object was placed at non-zero coordinates, skewing every child object attached to it.

After: 'Track' child object is now set to (0, 0, 0), allowing for easy uniform scaling.

jtbui20 committed:
Merge branch 'dev' into feature/rds-209-vehicle-movement-state-machine_extended

jtbui20 committed:
feat: Update defaults

jtbui20 committed:
fix: Fix MovementVector undeclared

jtbui20 committed:
feat: refactor jump handling

OoXLR8oO committed:
Small updates to race track

jtbui20 committed:
feat: Adjust implementation to allow hold release jumps again

jtbui20 committed:
Merge branch 'feature/rds-209-vehicle-movement-state-machine_extended' into rds-23-course-iterations-1

OoXLR8oO committed:
Merge branch 'dev' into rds-23-course-iterations-1

jtbui20 committed:
feat: Add grinds to map

AppelJoosh committed:
remove redundant spline

Spline object had two splines in its container, one was pointless(?), removed it

OoXLR8oO committed:
Merge remote-tracking branch 'origin/feature/rds-209-vehicle-movement-state-machine_extended' into rds-23-course-iterations-1

OoXLR8oO committed:
Merge branch 'rds-23-course-iterations-1' of https://github.com/team-riders/project-riders into rds-23-course-iterations-1

jtbui20 committed:
feat: Implement momentum & gravity to grind, implement jump exit

jtbui20 committed:
fix: Make turning more lurchy again

jtbui20 committed:
Merge branch 'feature/rds-209-vehicle-movement-state-machine_extended' into rds-23-course-iterations-1

jtbui20 committed:
Merge remote-tracking branch 'origin/dev' into rds-23-course-iterations-1
